demos/machine_translation_demo/python/machine_translation_demo.py

```diff
@@ -204,18 +210,24 @@ def main(args):
     model = Translator(
         model_xml=args.model,
         model_bin=args.model.with_suffix(".bin"),
+        device=args.device,
         tokenizer_src=args.tokenizer_src,
         tokenizer_tgt=args.tokenizer_tgt,
         output_name=args.output_name
     )
-
-    if args.input:
-        def sentences():
-            for sentence in args.input:
+    input_data = args.input
+    if input_data and len(input_data) == 1 and input_data[0].endswith('.txt'):
+        input_data = open(input_data[0], 'r', encoding='utf8')
+    if args.output:
+        with open(args.output, 'w') as f:
+            f.truncate()
+
+    def sentences():
+        if input_data:
+            for sentence in input_data:
```

Comment:
If `input_data` is a file, then `sentence` will contain a trailing newline. You should probably strip it.